src/app/flows/fines/fines-acc/fines-acc-debtor-add-amend/fines-acc-debtor-add-amend-form/fines-acc-debtor-add-amend-form.component.ts

```diff
@@ -0,0 +1,258 @@
+import { nationalInsuranceNumberValidator } from '@hmcts/opal-frontend-common/validators/national-insurance-number';
+import { optionalMaxLengthValidator } from '@hmcts/opal-frontend-common/validators/optional-max-length';
+import { optionalValidDateValidator } from '@hmcts/opal-frontend-common/validators/optional-valid-date';
+import { GovukSelectComponent } from '@hmcts/opal-frontend-common/components/govuk/govuk-select';
+import { GovukTextInputComponent } from '@hmcts/opal-frontend-common/components/govuk/govuk-text-input';
+import { IGovUkSelectOptions } from '@hmcts/opal-frontend-common/components/govuk/govuk-select/interfaces';
+import { CapitalisationDirective } from '@hmcts/opal-frontend-common/directives/capitalisation';
+import { patternValidator } from '@hmcts/opal-frontend-common/validators/pattern-validator';
+import { optionalPhoneNumberValidator } from '@hmcts/opal-frontend-common/validators/optional-valid-telephone';
+import {
+  ALPHANUMERIC_WITH_SPACES_PATTERN,
+  LETTERS_WITH_SPACES_PATTERN,
+  EMAIL_ADDRESS_PATTERN,
+} from '@hmcts/opal-frontend-common/constants';
+import { GovukCancelLinkComponent } from '@hmcts/opal-frontend-common/components/govuk/govuk-cancel-link';
+
+// regex pattern validators for the form controls
+const LETTERS_WITH_SPACES_PATTERN_VALIDATOR = patternValidator(LETTERS_WITH_SPACES_PATTERN, 'lettersWithSpacesPattern');
+const ALPHANUMERIC_WITH_SPACES_PATTERN_VALIDATOR = patternValidator(
+  ALPHANUMERIC_WITH_SPACES_PATTERN,
+  'alphanumericTextPattern',
+);
+const EMAIL_ADDRESS_PATTERN_VALIDATOR = patternValidator(EMAIL_ADDRESS_PATTERN, 'emailPattern');
+
+@Component({
+  selector: 'app-fines-acc-debtor-add-amend-form',
+  imports: [
+    FormsModule,
+    ReactiveFormsModule,
+    GovukButtonComponent,
+    GovukErrorSummaryComponent,
+    GovukSelectComponent,
+    GovukCheckboxesComponent,
+    GovukCheckboxesItemComponent,
+    GovukCheckboxesConditionalComponent,
+    GovukTextInputComponent,
+    GovukRadioComponent,
+    GovukRadiosItemComponent,
+    GovukCancelLinkComponent,
+    MojDatePickerComponent,
+    MojTicketPanelComponent,
+    CapitalisationDirective,
+  ],
+  templateUrl: './fines-acc-debtor-add-amend-form.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class FinesAccDebtorAddAmendFormComponent extends AbstractFormAliasBaseComponent implements OnInit, OnDestroy {
+  private readonly finesAccountStore = inject(FinesAccountStore);
+
+  @Output() protected override formSubmit = new EventEmitter<IFinesAccDebtorAddAmendForm>();
+  protected readonly dateService = inject(DateService);
+  protected readonly finesAccRoutingPaths = FINES_ACC_DEFENDANT_ROUTING_PATHS;
+  protected readonly showLanguagePreferences = computed(() => this.finesAccountStore.welsh_speaking() === 'Y');
+
+  @Input({ required: true }) public partyType!: string;
+  @Input({ required: false }) public initialFormData!: IFinesAccDebtorAddAmendForm;
```

Comment:
If you do the following, if initialFormData isn't supplied it will get the constant value which allows you to ignore the const in the `initialDebtorAddAmendSetup()` method and you can just pass `this.rePopulateForm(this.initialFormData.formData);`

```suggestion
  @Input({ required: false }) public initialFormData: IFinesAccDebtorAddAmendForm = FINES_ACC_DEBTOR_ADD_AMEND_FORM;
```

Reply:
Nevermind yep you are right